packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -283,7 +285,8 @@ export class DataStores implements IDisposable {
         const context = await this.contexts.getBoundOrRemoted(id, wait);
 
         if (context === undefined) {
-            throw new Error(`DataStore ${id} does not yet exist or is not yet bound`);
+            // The requested data store does not exits. Throw a 404 response exception.
+            throw responseToException(create404Response());
```

Comment:
It might be useful to provide a bit more info for this error to contain.
Either a request (to be created on the fly) that has URI (that can be used later for debugging) or test (see create404Response implementation - it's one liner, we can createResponseError and provide explanation text)

Note that ID should not be part of text, but it can be part of request. Reason being - we can't output it to telemetry right now as we did not solve PII concerns, and request.uri is not put into telemetry, but as we solve them, we will likely collect it when it's safe, and thus you will get it for free if you use request here.

Reply:
That's what I started with - request with the id. But I was unsure about whether that will be filtered because its PII.
Good to know, I will add a request here.